### ISSUE TITLE
Stacked line chart with negative values

### ISSUE THREAD

**Issue by coettl** [apache/echarts#9317]
### One-line summary [问题简述]

When i have a line with some negative values (e.g. [3, 1, 2, **-1, -3,** 1, 3]) and my stack is always 1, i want that the stack also work for "-1" and "-3".

### Version & Environment [版本及环境]
+ ECharts version [ECharts 版本]: 4
+ Browser version: all browsers
+ OS Version: all OS

### ECharts option [ECharts配置项]

```javascript
option = {
    xAxis : [
        {
            type: 'category',
        		data: [1,2,3,4,5,6,7,8,9]
        }
    ],
    yAxis : [
        {
            type : 'value'
        }
    ],
    series : [
        {
            name:'stack',
            type:'line',
            stack: 'stack',
            data:[3, 1, 2, -1, -3, 1, 3]
        },
        {
            name:'stack',
            type:'line',
            stack: 'stack',
            areaStyle: {},
            data:[1, 1, 1, 1, 1, 1, 1]
        }
    ]
};
```

### Expected behaviour [期望结果]
I get the following result with the option above.

![image](https://user-images.githubusercontent.com/33359063/47794370-9e571800-dd20-11e8-9bed-7b8642280600.png)

But i would expect something like this:
![image](https://user-images.githubusercontent.com/33359063/47794757-61d7ec00-dd21-11e8-9b9b-f706865c788e.png)

### Other comments [其他信息]

Example of the bug
https://jsfiddle.net/ykb5n0pa/

**Comment by dmichlmayr** [comment]
+1

**Comment by 100pah** [comment]
Currently, the stack rule is "positive values stack with positive values, and negative values stack with negative values".

I guess actually your need an "area chart"?
But unfortunately, this type of area chart has not been supported yet. 
It should be scheduled.

**Comment by MINDoSOFT** [comment]
Hi everyone!

This would be a very useful feature to have !

Currently you can bypass this problem, by having one series with all the negative values and one series with all the positive values (sharing the same series name). I added an [example here as reference](https://jsfiddle.net/758vs6xe/1/). You get an empty space when you switch between the negative and the positive part of the series, but I couldn't find a way to address this.

Best regards,
Sergios.

**Comment by h4de5** [comment]
btw - we worked around this issue, by drawing a single polygon as custom shape into chart.

as an example we used this code: https://stackoverflow.com/questions/61724715/echarts-plot-the-variance-of-signals

**Comment by villebro** [comment]
@Ovilia @100pah this is also a feature I would need, and I would be happy to work on it. Is this something the project would be open to merging if I open a PR that maintains backward compatibility but adds this as an alternative option? If so, what would be the preferred API change for enabling this? Also, if there's a good way of working around this I'm happy to add docs/examples so others can do this more easily in the future.

**Comment by pissang** [comment]
@villebro It will be great! Thanks for picking this up. I think proving a config like stackStrategy is an option.

**Comment by villebro** [comment]
@pissang Perfect, I'll get to work on it right away!

**Comment by nebojsa-zakic** [comment]
We need this feature as well. Is there any progress perhaps?

**Comment by bigfacewo** [comment]
Any update?

**Comment by villebro** [comment]
Hi all, sorry for the delay. I have the feature branch ready, but I got sidetracked before I was able to finalize it (tests etc). I also need this now, so I'll do my best to get the PR opened in the next few days!

**Comment by nelsonbbarriento23** [comment]
Isbm???nhunn^<

**Comment by nelsonbbarriento23** [comment]
Being able to bring jr

**Comment by villebro** [comment]
Here's the proposed fix PR: #17086
